src/MassTransit/Consumers/Batching/BatchConsumer.cs

```diff
@@ -197,20 +202,20 @@ async Task Deliver(ConsumeContext context, IReadOnlyList<ConsumeContext<TMessage
 
         List<ConsumeContext<TMessage>> GetMessageBatchInOrder()
         {
-            return _messages.Values.OrderBy(x => x.Timestamp).Select(x => x.Context).ToList();
+            return _messages.Values.OrderBy(x => x.Ordinal).Select(x => x.Context).ToList();
         }
 
 
         readonly struct BatchEntry
         {
             public readonly ConsumeContext<TMessage> Context;
-            public readonly DateTime Timestamp;
+            public readonly ulong Ordinal;
             readonly CancellationTokenRegistration _registration;
 
-            public BatchEntry(ConsumeContext<TMessage> context, DateTime timestamp, Action canceled)
+            public BatchEntry(ConsumeContext<TMessage> context, ulong ordinal, Action canceled)
```

Comment:
and why not just use `sequenceNumber` instead of `ordinal` ?

Reply:
Or index?

Reply:
The SQL transport also has a sequence number

Reply:
my thinking was this wasn't a sequenceNumber in the fallback case.  Overthinking it.  will change

Reply:
> The SQL transport also has a sequence number

DeliveryMessageId?